src/drivers/drivers.jl

```diff
@@ -85,9 +86,24 @@ function set_driver!(p::Properties, ::POSIX)
 end
 
 function __init__()
-    DRIVERS[API.h5fd_core_init()] = Core
-    DRIVERS[API.h5fd_sec2_init()] = POSIX
-    @require MPI="da04e1cc-30fd-572f-bb4f-1f8673147195" include("mpio.jl")
+    # disable file locking as that can cause problems with mmap'ing
+    if !haskey(ENV, "HDF5_USE_FILE_LOCKING")
```

Comment:
What's the reason we need to add this check here as well? Is it related to `Drivers` being in its own module?

Reply:
What we really need to do is address #959. Basically what happens is that the `Drivers` module is initialized before the `HDF5` module. However, if we create a FAPL during `Drivers` initialization, then the environment variable no longer has the desired effect. Then memory mapping breaks before the files are now locked.

Reply:
Thanks that makes sense.